### PR TITLE
Feature/request handler

### LINKED ISSUE #20: Implement Request Handler
**Is your feature request related to a problem? Please describe.**
The is currently no `RequestHandler` implemented. Without we can use the `DroneController` at all.

**Describe the solution you'd like**
There is a description in the [documentation](https://treeandsea.github.io/DroneController/). But a brief summary: It expects different dictionaries ranging from user input to an expected `DroneState`. It has an optional parameter for redirect url.

- [ ] handle keyboard input

- [ ] handle game pad input

- [ ] handle drone state input

### PR DESCRIPTION
**Pull request by Segelzwerg**
This implements part of the request handler.
The request handler will be able to pass keyboard input to the thrust calculator and request calculation.

closes #20

### CODE REVIEW

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# pylint: disable=too-few-public-methods
+class ThrustCalculator:
```
Comment (TreeKid) [inline src/drone_controller/calculation_layer/thrust_calculator.py]: I think the ThrustCalculator should be Drone specific. eg. ThrustCalculatorQuatrocopter.
I believe calculating the correct thrust highly depends on the drone setup.

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    def __init__(self, drone_state, expected_state):
+        """
+        :param drone_state: the momentary drone state
+        :param expected_state: the expected state of the dron
+        """
+        self.drone_state = drone_state
+        self.expected_state = expected_state
+
+    def calc(self, rotor_count):
```
Comment (TreeKid) [inline src/drone_controller/calculation_layer/thrust_calculator.py]: I doubt, that you can use the same calculation for different number of rotors.
Physic is usually not that simple :P

File: src/drone_controller/input_layer/request_handler.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+"""
+This module handles all requests. This will cover REST calls as well as native python calls.
+"""
+from src.drone_controller.calculation_layer.thrust_calculator import ThrustCalculator
+from src.drone_controller.exception.exceptions import UserInputError
+from src.drone_controller.input_layer.drone_state import DroneState
+from src.drone_controller.input_layer.drone_state_mapper import DroneStateMapper
+
+
+class RequestHandler:
```
Comment (TreeKid) [inline src/drone_controller/input_layer/request_handler.py]: With this class setup, every step you call `RequestHandler.keyboard_input(x, y)` and create a new Instance of RequestHandler, DroneStateMapper and ThrustCalculator.

Can't you init all classes once and than siply call their methods?

File: src/drone_controller/input_layer/request_handler.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                                                 "Rotation Left",
+                                                 "Acceleration"]):
+            raise UserInputError(user_input, "The keyboard input is complete.")
+
+        state_mapper = DroneStateMapper()
+        expected_state = state_mapper.keyboard(user_input)
+        return self.calc(drone_state, expected_state)
+
+    def calc(self, drone_state, expected_state):
+        """
```
Comment (Segelzwerg) [inline src/drone_controller/input_layer/request_handler.py]: Codacy Issue found: [Multi-line docstring summary should start at the first line](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state.py  1
- src/drone_controller/input_layer/drone_state_mapper.py  1
- src/drone_controller/exception/exceptions.py  1
- src/drone_controller/input_layer/request_handler.py  3
- src/drone_controller/calculation_layer/thrust_calculator.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179&bid=17072433)

File: src/drone_controller/input_layer/drone_state.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# pylint: disable=too-few-public-methods
+# keep this only until this is implemented
+
+
+class DroneState:
+    """
```
Comment (Segelzwerg) [inline src/drone_controller/input_layer/drone_state.py]: Codacy Issue found: [Multi-line docstring summary should start at the first line](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+from src.drone_controller.input_layer.drone_state import DroneState
+
+
+class ThrustCalculatorQuadroCopter:
+    """
+    Calculator for rotor thrusts.
+    """
+
+    def calc(self, drone_state: DroneState, expected_state: DroneState):
+        """
```
Comment (Segelzwerg) [inline src/drone_controller/calculation_layer/thrust_calculator.py]: Codacy Issue found: [Multi-line docstring summary should start at the first line](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/exception/exceptions.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    Base class for exception in this module.
+    """
+
+
+class UserInputError(DroneControllerError):
+    """
+    Incorrect user input.
+    """
+
+    def __init__(self, expression, message):
```
Comment (Segelzwerg) [inline src/drone_controller/exception/exceptions.py]: Codacy Issue found: [Missing docstring in __init__](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/input_layer/drone_state.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# pylint: disable=too-few-public-methods
```
Comment (Segelzwerg) [inline src/drone_controller/input_layer/drone_state.py]: Codacy Issue found: [Missing module docstring](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/exception/exceptions.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+class DroneControllerError(Exception):
+    """
+    Base class for exception in this module.
+    """
+
+
+class UserInputError(DroneControllerError):
+    """
```
Comment (Segelzwerg) [inline src/drone_controller/exception/exceptions.py]: Codacy Issue found: [Multi-line docstring summary should start at the first line](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/input_layer/request_handler.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+from src.drone_controller.input_layer.drone_state import DroneState
+from src.drone_controller.input_layer.drone_state_mapper import DroneStateMapper
+
+
+class RequestHandler:
+    """
+    Handles request from python code.
+    """
+
+    def __init__(self, aircraft_type):
```
Comment (Segelzwerg) [inline src/drone_controller/input_layer/request_handler.py]: Codacy Issue found: [Missing docstring in __init__](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/exception/exceptions.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+class DroneControllerError(Exception):
```
Comment (Segelzwerg) [inline src/drone_controller/exception/exceptions.py]: Codacy Issue found: [Missing module docstring](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/input_layer/drone_state_mapper.py [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+# pylint: disable=too-few-public-methods
+# keep this only until this is implemented
+class DroneStateMapper:
+    """
+    Maps different inputs to an expected drone state.
+    """
+
+    def keyboard(self, user_input):
+        """
```
Comment (Segelzwerg) [inline src/drone_controller/input_layer/drone_state_mapper.py]: Codacy Issue found: [Multi-line docstring summary should start at the first line](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

File: src/drone_controller/calculation_layer/thrust_calculator.py [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# pylint: disable=too-few-public-methods
```
Comment (Segelzwerg) [inline src/drone_controller/calculation_layer/thrust_calculator.py]: Codacy Issue found: [Missing module docstring](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/request_handler.py  3
- src/drone_controller/calculation_layer/thrust_calculator.py  1
- src/drone_controller/input_layer/drone_state_mapper.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179&bid=17072433)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/request_handler.py  3
- src/drone_controller/calculation_layer/thrust_calculator.py  1
- src/drone_controller/input_layer/drone_state_mapper.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179&bid=17072433)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/request_handler.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5205179&bid=17072433)

**Pull request merged**